src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -1855,6 +1857,63 @@ int CpuemShim::xclCloseContext(const uuid_t xclbinId, unsigned int ipIndex) cons
   return 0;
 }
 
+
+int CpuemShim::xclIPName2Index(const char *name)
+{ 
+  std::string errmsg;
+  const uint64_t bad_addr = 0xffffffffffffffff;
+  char* ipLayoutbuf = nullptr;
+  size_t bufsize = 0;
+
+  if (!mHeader) {
+    errmsg = "ERROR: [SW-EMU 13] xclIPName2Index - Invalid xclbin content";
+    std::cout << errmsg << std::endl;
+    return -EINVAL;
+  }
+
+  char *bitstreambin = reinterpret_cast<char*> (mHeader);
+  auto top = reinterpret_cast<const axlf*>(mHeader);
+  if (auto sec = xclbin::get_axlf_section(top, IP_LAYOUT)) {
+    bufsize = sec->m_sectionSize;
+    ipLayoutbuf = new char[bufsize];
```

Comment:
Why a copy, (see next later comment).

Reply:
Removed code for IP_LAYOUT copy to ipLayoutbuf